Added support for Gigabyte AX370 Gaming K7 (Added Partial-Ryzen Support Too)

Kraust opened the pull request:
![AX370](http://i.imgur.com/qTf4Y6W.png)

I have not figured out how to get Ryzen's Temperature / Vcore sensors yet as AMD hasn't released their guide on how to do this. Also, some of the scalars for the voltages (e.g. +12v) I reverse calculated based on HWiNFO's data.

DarthDestroyer commented:
Good Job!

vinnier commented:
Can confirm these changes work on my Gigabyte AB350 Gaming 3 since it uses the same chip. If you'd like to add support for my motherboard, its string id is "AB350-Gaming 3-CF". Otherwise, I'll create a pull request once these changes are in.

lucascosti commented:
Great job, this looks to be working with my Z270X-Gaming K5 board too :)

myfreeweb commented:
This "Ryzen support" (same as #962) does not read the actual frequency, it reads the stock frequency :( It's always 3 GHz on my 1700, when it's actually running at 3.92

berendiwema commented:
Hello @Kraust , 

Do you have the compiled DLL's by any chance? I'm hoping to drop them in with the Sidebar Gadget app and get CPU temps :)

With kind regards,
Berend

Hubfront commented:
Looks interesting. Where can i download the compiled binary?

DarthDestroyer commented:
Only a little advice to anyone who asks for compiled software:

Only download from trusted sources!
Self compiling is so hard as writing a Word Document... ;)

bigpodgurc commented:
Yea specaily c#

Kraust commented:
Sorry I tried to do some updating of VS yesterday, and I was getting some very weird errors. My Windows install must have gotten borked somehow and I don't have access to VS right now.

Hubfront commented:
"Self compiling is so hard as writing a Word Document... ;)"

Probably not for me. ;-)

I installed visual studio community from
https://www.visualstudio.com/de/thank-you-downloading-visual-studio/?sku=Community&rel=15#
but unfortunately i did not manage to build an exe yet from the project file OpenHardwareMonitor.csproj .

Does a help page exist as starting point for do it yourself compiling?

DarthDestroyer commented:
#995 was the last issue i had explained it further... hope that helps!

Hubfront commented:
Hi Kraust, thank you for your reply. I will try do compile it by myself.

Hubfront commented:
"#995 was the last issue i had explained it further... hope that helps!"

Thank you, ill try !

berendiwema commented:
@DarthDestroyer cheers, compiled the DLL, works as expected.

DarthDestroyer commented:
perfect! that is the best way and im happy that i could have helped!

Kraust commented:
Updated 2017/11/12:

I added additional support for Ryzen temperature monitoring and several new chipsets. I will go through the comments now and see if there's anything else I can add as well.

I also want to tackle the "RX 400 and RX 500 series GPUs do not report Voltage Properly" issue, but I know there were recent driver changes on AMD's side which might impact this.

TwistedMetal85 commented:
Where I cant find .exe with ryzen support ??

DarthDestroyer commented:
@TwistedMetal85 There is NO Executable available atm, that will support it! But, you can do it yourself by visiting #995 , where it is further explained how you could get an "exe" ;)

myfreeweb commented:
The 20 degree offset is only on chips with XFR (the X models), don't add it for everyone

t0mb3r commented:
Hello, Can someone upload the compiled dll with the ryzen support?

t0mb3r commented:
ok, so I have compiled the software and it works with some issues, for some reason the CPU frequency isnt shown correctly, it shows the stock clock, not the actual clock, and the CPU temperature has an odd 49 degrees offset. 
![image](https://user-images.githubusercontent.com/40329189/44667406-150a2900-a9f1-11e8-8458-ebb6f8893378.png)

t0mb3r commented:
@TwistedMetal85 thank you but I need this version because I use it with other software

t0mb3r commented:
@Kraust I have done some progress, I figured that the temperature is really an odd offset and I was able to display it correctly,  now the CPU CLOCK is not displayed correctly, is there a workaround? My motherboard is AB350M-D3H, I have added it.

TwistedMetal85 commented:
@t0mb3r is a 0.8.0.2 version but someone fix it. 
![ohm ryzen](https://user-images.githubusercontent.com/35374409/44688455-5109b200-aa22-11e8-9c24-7fb5aa5f266f.JPG)

Have you tried my .zip file?

t0mb3r commented:
Yes, the one you are using is based on the librehardwaremonitor, for some reason it doesnt work with what Im using, but somehow I have everything running now, there is no offset with the temperature but the cpu frequency only shows correctly if the cpu is overclocked though the bios. Any ways, everything is working in most of the part. Thank you and it would be awesome to continue with the support.

SearchForTheCode commented:
I wanted to ask if there is any update in sight? I know you can use the librehardwaremon, but it is not matching our target .net framework .

I know i can manually compile the pull requests but as this issue is now older than one year, i am wondering if there will be updates at all to this project?

DarthDestroyer commented:
That is a question only @moellerm could answer!

I am sure, that there will be a update but as i said, only he could give an eta! 

A big help would be, that all Pull Request Tester give Feedback! As for sure, no one can get all kind of Hardware and its combinations... That would help a lot to get everything accepted!

SearchForTheCode commented:
Thanks for the fast answer! I will make sure that we get this pull request tested. I updated my code with these changes and try to get feedback within the next weeks and will report back!

mcarbonneaux commented:
i've compiled and tested this patch and it's work find with my ryzen 5 2600 + gigabyte b450 aorus!

SearchForTheCode commented:
i can also say that this patch is working! So it should be forked into the master branch :)

MasonLyons commented:
Just tested and it's reading 90C on my 2700x using an ASUS PRIME X470-PRO and Ryzen Master is reporting 31C

MasonLyons commented:
I've done some more testing and to get the right value as reported by Ryzen master I have to do this
`(sensor.value * 10) - 273.15) / 10`

SearchForTheCode commented:
Thats some good input! Was this just testing with the value or did you find any sort of documentation where this formula is based off?

MasonLyons commented:
It was after some trial and error when trying to convert the numbers between different units. It was clear the number reported was wrong so I just started messing around with it. I vaguely remember seeing something related to a multiplication of 10 in some other temperature program but I'm not confident where or if it was related. We're using it as part of an automation framework for reviews so until the issue gets fixed I'm manually applying this fix for AMD. The was specifically tested on a Z399 AORUS XTREME-CF with a Threadripper 1950x

DarthDestroyer commented:
it is not so far away for a sensor to include some kind of "zero"! At this point it is the absolute zero in Celsius or 0K. 

But if you overlook the math, it could be easier and shorter ;-)

SearchForTheCode commented:
Any update on this? 
I have some " AMD Ryzen 5 2400G CPU/APU" which seems to be not working with this commits! AMD Ryzen seems to be getting out of hands with all those new weird stuff.

aquacomputer commented:
can you try out the latest version: https://github.com/LibreHardwareMonitor/LibreHardwareMonitor

SearchForTheCode commented:
I will try it out if i get access to those machines again :)

SearchForTheCode commented:
Yes i tested it with one machine and can confirm it!! 
Any chance this will get included in OHM? Otherwise i need to do a analysis myself what part changed in Libre so i can do the changes in my local OHM source
